inc/zoo/swar/SWAR.h

```diff
@@ -62,18 +73,82 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+            if constexpr (NBits < sizeof(T) * 8) {
+                return (T(1) << NBits) - 1;
+            } else {
+                return ~T(0);
+            }
+        }(),
         // Use LowerBits in favor of ~MostSignificantBit to not pollute
         // "don't care" bits when non-power-of-two bit lane sizes are supported
-        LowerBits = MostSignificantBit - LeastSignificantBit;
+        LowerBits = MostSignificantBit - LeastSignificantBit,
+        MaxUnsignedLaneValue = LeastSignificantLaneMask;
+
+    template <typename InputIt>
+    constexpr static auto from_range(InputIt first, InputIt last) noexcept {
+        auto result = T{0};
+        for (; first != last; ++first) {
+            result = (result << NBits) | *first;
+        }
+        return result;
+    }
+
+    template <typename U>
+    constexpr static auto from_array(const U (&values)[Lanes]) noexcept {
+        using std::begin; using std::end;
+        return SWAR{from_range(begin(values), end(values))};
+    }
+
+    template <typename U>
+    constexpr static auto from_array(const std::array<T, Lanes> &values) noexcept {
+        using std::begin; using std::end;
+        return SWAR{from_range(begin(values), end(values))};
+    }
```

Comment:
I believe we can have a `from_range(Range` overload and not make it specific to `std::array` unnecessarily

Reply:
Yes, I think that would be good. Btw I didn't find the SWAR `begin` and `end` functions I think you mentioned in another comment. Was wondering if you could explain more about that?

Reply:
Not in SWAR, but they could be implemented... the thing is what ought to be the type of a SWAR iterators to iterate over all the lanes? I think iterating serially over the lanes is of low priority.
This overload should be turned into this:
```suggestion
    template <typename Range>
    constexpr static auto from(const Range &values) noexcept {
        using std::begin; using std::end;
        return SWAR{from_range(begin(values), end(values))};
    }
```

Reply:
I seee, nice.